src/transaction/components/TransactionProvider.tsx

```diff
@@ -73,17 +84,48 @@ export function TransactionProvider({
     hash: writeContractTransactionHash || transactionHash,
   });
 
+  const getTransactionReceipts = useCallback(async () => {
+    const receipts = [];
+    for (const hash of transactionHashArray) {
+      try {
+        const txnReceipt = await waitForTransactionReceipt(config, {
+          hash,
+          chainId,
+        });
+        receipts.push(txnReceipt);
+      } catch (err) {
+        console.error('getTransactionReceiptsError', err);
+        setErrorMessage(genericErrorMessage);
+      }
+    }
+    setReceiptArray(receipts);
+  }, [transactionHashArray]);
+
+  useEffect(() => {
+    if (transactionHashArray.length === contracts.length && contracts?.length > 1) {
+      getTransactionReceipts();
+    }
+  }, [contracts, transactionHashArray]);
+
   const fallbackToWriteContract = useCallback(async () => {
     // EOAs don't support batching, so we process contracts individually.
     // This gracefully handles accidental batching attempts with EOAs.
     for (const contract of contracts) {
       try {
-        await writeContract(contract);
-      } catch (_err) {
-        setErrorMessage(genericErrorMessage);
+        await writeContractAsync?.(contract);
+      } catch (err) {
+        // if user rejected request
+        if (
+          (err as TransactionExecutionError)?.cause?.name ===
+          'UserRejectedRequestError'
+        ) {
+          setErrorMessage('Request denied.');
+        } else {
+          setErrorMessage(genericErrorMessage);
+        }
       }
     }
-  }, [contracts, writeContract]);
+  }, [contracts, writeContractAsync]);
```

Comment:
please make sure to test this on mobile safari. We've had some issues with them blocking our popup with this async call

Reply:
@Zizzamia how would you recommend we test this on mobile safari ?